rtengine/jdatasrc.cc

```diff
@@ -409,24 +395,14 @@ reset_error_mgr (j_common_ptr cinfo)
 GLOBAL(struct jpeg_error_mgr *)
 my_jpeg_std_error (struct jpeg_error_mgr * err)
 {
+    err = jpeg_std_error(err);
 
+    /* override these functions */
     err->error_exit = my_error_exit;
     err->emit_message = emit_message;
     err->output_message = output_message;
     err->format_message = format_message;
     err->reset_error_mgr = reset_error_mgr;
```

Comment:
It seems to me `my_error_exit` is the only customized function. The other ones are exact copies of the ones in libjpeg-turbo and are already added to the `err` struct by `jpeg_std_error()`. Thus, the only thing this function (`my_jpeg_std_error`) needs to do is call `jpeg_std_error()` then override the `error_exit` function. The copied functions can be removed from this file.

```suggestion
    err = jpeg_std_error(err);

    /* override these functions */
    err->error_exit = my_error_exit;
```